net/dns/nrpt_windows.go

```diff
@@ -348,28 +316,25 @@ func (db *nrptRuleDatabase) refreshLocked() {
 }
 
 func (db *nrptRuleDatabase) writeNRPTRule(ruleID string, servers, doms []string) error {
-	var nrptBase string
-	if db.writeAsGP {
-		nrptBase = nrptBaseGP
-	} else {
-		nrptBase = nrptBaseLocal
+	subKeys := []string{nrptBaseLocal, nrptBaseGP}
+	if !db.writeAsGP {
+		subKeys = subKeys[:1]
```

Comment:
It might be clearer to do this as an `append` if `writeAsGP` is set, but I suppose this makes it easier to avoid an allocation.

Reply:
Yeah, that's how I originally had it, and then changed it to remove the allocation. Perhaps I'll opt for clarity in this case, since this is essentially a micro-optimization.

Reply:
```golang
subkeys := make([]string, 0, 2)
subkeys = append(subkeys, nrptBaseLocal)
if db.writeAsGP {
  subkeys = append(subkeys, nrptBaseGP)
}
```
🤮

Reply:
It's fine how it is, there's 2 entries so it's not hard to figure out. The most I'd change is adding a comment.